docs/cookbook/promotions/catalog-promotion-additional-item-filters.rst

```diff
@@ -0,0 +1,157 @@
+How to add custom filters on the item promotion actions?
+========================================================
+
+The "Item percentage discount" and "Item fixed discount" promotion actions can be used to filter which products the promotion is applied to.
+By default, you can filter by price, by taxon or by product.
+
+You can also add your own filters, for example, if you've added a brand to your products and would like to apply a discount to a particular brand.
+
+See what steps need to be taken to achieve that:
+
+Create a new item filter
+------------------------
+
+You will need a new class ``BrandFilter`` that implements the ``Sylius\Component\Core\Promotion\Filter\FilterInterface`` interface.
+
+The `filter` method will check that items match the criteria from the `$configuration` and returns only those that do.
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Promotion\Filter;
+
+    use Sylius\Component\Core\Model\ProductInterface;
+    use Sylius\Component\Core\Promotion\Filter\FilterInterface;
+
+    final class BrandFilter implements FilterInterface
+    {
+        public const EQUAL_CONDITION = 'eq';
+        public const NOT_EQUAL_CONDITION = 'neq';
+
+        public function filter(array $items, array $configuration): array
+        {
+            if (empty($configuration['filters']['brand_filter']['brand'])) {
+                return $items;
+            }
+
+            $brandValue = $configuration['filters']['brand_filter']['brand'];
+            $brandCondition = $configuration['filters']['brand_filter']['brand_condition'] ?? self::EQUAL_CONDITION;
```

Comment:
Extra condition is too complex and IMO not necessary for this example, copy of the product/taxon filter should suite better to show-case it.